File: frontend/src/app/models/optionValue.ts
import { OptionList } from './optionList';
export class OptionValue {
    
    
    idx: number;
    optionListId: number;
    optionList?: OptionList;
    value: string;
    
    constructor(data:any) {
        this.idx = data.idx;
        this.optionListId = data.optionListId;
        this.optionList = data.optionList;
        this.value = data.value;
    }
}

File: backend/Models/OptionValue.cs
using CsvHelper.Configuration;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace prid_2425_a01.Models;


public class OptionValue {

    public int Idx { get; set; }
    public int OptionListId { get; set; }
    public string Value { get; set; } = "";
    
    [ForeignKey(nameof(OptionListId))]
    //[NotMapped]
    public OptionList OptionList { get; set; } = null!;
}
